snarkos/store/rocksdb/tests.rs

```diff
@@ -14,185 +14,170 @@
 // You should have received a copy of the GNU General Public License
 // along with the snarkOS library. If not, see <https://www.gnu.org/licenses/>.
 
-use crate::storage::{rocksdb::RocksDB, DataID, DataMap};
-
-use snarkvm::{
-    compiler::{Map, MapRead},
-    console::{
-        network::{Network, Testnet3},
-        types::Scalar,
-    },
-    utilities::{test_crypto_rng, Uniform},
+use crate::{
+    store::rocksdb::{DataMap, Database, RocksDB},
+    DataID,
 };
+use serial_test::serial;
+use snarkvm::compiler::{Map, MapRead};
 
-// use snarkvm::compiler::{Map, MapRead};
+use std::fs;
 
-pub(crate) fn temp_dir() -> std::path::PathBuf {
-    tempfile::tempdir().expect("Failed to open temporary directory").into_path()
+fn remove_test_dir(network_id: u16) {
+    let _ = fs::remove_dir_all(format!("~/.aleo/storage/.ledger-{}", network_id));
 }
 
-pub(crate) fn temp_file() -> std::path::PathBuf {
-    tempfile::NamedTempFile::new()
-        .expect("Failed to open temporary file")
-        .path()
-        .to_owned()
+fn setup_test_map(network_id: u16) -> TestMap {
+    remove_test_dir(network_id);
+    RocksDB::open_map(network_id, DataID::Test).expect("Failed to open data map")
 }
 
+type TestMap = DataMap<u32, String>;
+
 #[test]
+#[serial]
 fn test_open() {
-    let _storage = RocksDB::open(temp_dir(), 0).expect("Failed to open storage");
+    remove_test_dir(0);
+    assert!(RocksDB::open(0).is_ok());
 }
 
 #[test]
-fn test_open_map() {
-    let storage = RocksDB::open(temp_dir(), 0).expect("Failed to open storage");
-    storage.open_map::<u32, String>(DataID::Test).expect("Failed to open data map");
+#[serial]
+fn test_open_map_opens_a_datamap_for_a_given_dataid() {
+    remove_test_dir(1);
+    assert!(RocksDB::open_map::<i32, String>(1, DataID::Test).is_ok());
 }
 
-type TestMap = DataMap<u32, String>;
+#[test]
+#[serial]
+fn test_a_key_is_contained_after_a_value_was_inserted_with_it() {
+    let test_map = setup_test_map(2);
+
+    test_map.insert(123456789, "123456789".to_string()).expect("Failed to insert");
+    let expected_result = test_map.contains_key(&123456789);
+
+    assert!(expected_result.is_ok());
+}
 
 #[test]
-fn test_insert_and_contains_key() {
-    let storage = RocksDB::open(temp_dir(), 0).expect("Failed to open storage");
-    let mut map: TestMap = storage.open_map(DataID::Test).expect("Failed to open data map");
+#[serial]
+fn test_a_key_is_not_contained_if_no_value_was_inserted_with_it() {
+    let test_map = setup_test_map(3);
```

Comment:
with the current setup, any test using a map with ID `3` will remove the storage for the (non-test) `testnet3`

Reply:
You're right!

Reply:
What do you think about receiving a `dev` parameter in the `open` method and replace the `None` later used in the `aleo_ledger_dir` call?

Reply:
it's an option, though IMO the best way would be to alter `aleo_std::aleo_ledger_dir` to use something like the `tempfile` crate in tests (note: the `dev` param in `aleo_ledger_dir` indicates dev mode, which is *not* the same as a test